builder/builder-next/exporter/export.go

```diff
@@ -73,8 +72,21 @@ func (e *imageExporterInstance) Name() string {
 	return "exporting to image"
 }
 
-func (e *imageExporterInstance) Export(ctx context.Context, ref cache.ImmutableRef, opt map[string][]byte) (map[string]string, error) {
-	if config, ok := opt[exporterImageConfig]; ok {
+func (e *imageExporterInstance) Export(ctx context.Context, inp exporter.Source) (map[string]string, error) {
+
+	if len(inp.Refs) > 1 {
+		return nil, fmt.Errorf("exporting multiple references to image store is currently unsupported")
+	}
+	ref := inp.Ref
+	if ref != nil && len(inp.Refs) == 1 {
```

Comment:
Was wondering why not `>= 1`, but you're taking the `if len(inp.Refs) > 1 {` into account. Perhaps more defensive to use `>= 1` here, but not a blocker

Really nitty; check `inp.Ref` here, instead of first assigning it to `ref`;

```go
if inp.Ref != nil && len(inp.Refs) == 1 {
	return nil, fmt.Errorf("invalid exporter input: Ref and Refs are mutually exclusive")
}
...
```